src/model/guild/guild_id.rs

```diff
@@ -195,13 +196,33 @@ impl GuildId {
     /// Also can return [`Error::Http`] if the current user lacks permission.
     ///
     /// [Ban Members]: Permissions::BAN_MEMBERS
-    pub async fn ban(self, http: &Http, user: UserId, dmd: u8, reason: Option<&str>) -> Result<()> {
-        Maximum::DeleteMessageDays.check_overflow(dmd.into())?;
+    pub async fn ban(
+        self,
+        http: &Http,
+        user: UserId,
+        delete_message_seconds: u32,
+        reason: Option<&str>,
+    ) -> Result<()> {
+        // Convert to usize for check overflow
+        const SEVEN_DAYS_MAXIMUM: u32 = 7 * 24 * 60 * 60;
+
+        if delete_message_seconds > SEVEN_DAYS_MAXIMUM {
+            #[cfg(target_pointer_width = "16")]
+            const {
+                panic!("Serenity is not supported on 16 bit platforms.")
+            };
+
+            return Err(Error::Model(ModelError::TooLarge {
+                maximum: Maximum::DeleteMessageSeconds,
+                value: SEVEN_DAYS_MAXIMUM as usize,
```

Comment:
1. This is meant for `delete_message_days`, not the maximum that was exceeded.
2. If you are doing `as` with the const panic you might as well use `.try_into().unwrap()`.

Reply:
Regarding point 1, I'm using Maximum::DeleteMessageSeconds here though (?) which should be the maximum exceeded here? Point 2 though, yeah, will fix

Reply:
Yes, the maximum exceeded is stored in the `maximum` field. The value that exceeded it is stored in `value`. You have passed the maximum (`SEVEN_DAYS_MAXIMUM`) into the `value` field.